project/BinaryIncompatibilities.scala

```diff
@@ -28,8 +28,40 @@ object BinaryIncompatibilities {
   val TestAdapter = Seq(
   )
 
+  private val JSTupleUnapplyExclusions: Seq[ProblemFilter] = {
+    /* !!! Very delicate
+     *
+     * We changed the result type of `js.TupleN.unapply` from `Option` to `Some`,
+     * to make them irrefutable from Scala 3's point of view. This breaks binary
+     * compat, so we added a `protected` overload with the old binary signature.
+     * Unfortunately, those do not get a *static forwarder*, and hence MiMa
+     * still complains about them. Although the error message is clearly about
+     * "static method"s, the *filter* to apply is indistinguishable between the
+     * instance and static methods! That means that *if* we messed up the
+     * instance method, these filters will filter their problems as well.
+     *
+     * To mitigate this, I manually followed these steps:
+     *
+     * 1. Change the signatures to the new ones, notice both static and
+     *    non-static incompatibilities.
+     * 2. Add the `protected` methods, notice that only static incompatibilites
+     *    remain.
+     * 3. Use scalajsp to double-check that the instance signatures are correct.
+     * 4. Only then add the problem filters here.
+     *
+     * Note that even though MiMa reports potential issues with static methods,
+     * these are ghost proplems. They do not exist in the .sjsir files to begin
+     * with, because the companion trait is a JS trait. We only generate static
+     * forwarders in Scala classes and traits. So filtering out the static
+     * method incompatibilities is legit.
+     */
```

Comment:
Instead of this, just write your own problem filter?

https://github.com/lightbend/mima/blob/main/core/src/main/scala/com/typesafe/tools/mima/core/package.scala

Something like:

```
def isStaticTupleUnapply(n: Int)(problem: Problem): Boolean = problem match {
  case problem: IncompatibleResultTypeProblem =>
    problem.ref.isStatic && problem.ref.fullName == s"scala.scalajs.js.Tuple$n.unapply"
  case _ => false
}
```

Reply:
Huh, I did not know this was even possible. Good idea.
Updated.